native/swift/Sources/wordpress-api/SafeRequestExecutor.swift

```diff
@@ -94,6 +101,16 @@ public final class WpRequestExecutor: SafeRequestExecutor {
         #endif
     }
 
+#if PROGRESS_REPORTING_ENABLED
+    public func progress(forRequestWithId requestId: String) -> AnyPublisher<Progress, Never> {
+        NotificationCenter.default.publisher(for: RequestExecutorDelegate.didCreateTaskNotification)
```

Comment:
IINM, apps should call this method then use `.receive(on: DispatchQueue.main)` right?

Reply:
The app does not need to call this method. It's declared `public` only because the `SafeRequestExecutor` is public.

The app should call the new `WordPressAPI.uploadMedia(...)` function if they want to observe the progress. They can call the usual `WordPressAPi.media.create(...)` function if they don't,